tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -46,201 +46,231 @@ use {
     },
 };
 
-#[derive(Error, Debug)]
-pub enum TpuSenderError {
-    #[error("Pubsub error: {0:?}")]
-    PubsubError(#[from] PubsubClientError),
-    #[error("RPC error: {0:?}")]
-    RpcError(#[from] ClientError),
-    #[error("IO error: {0:?}")]
-    IoError(#[from] std::io::Error),
-    #[error("Signer error: {0:?}")]
-    SignerError(#[from] SignerError),
-    #[error("Custom error: {0}")]
-    Custom(String),
-}
+pub mod temporary_pub {
```

Comment:
In order to deduplicate code between solana-tpu-client and solana-client, somethings needed to be made public. I wrapped them in these `temporary_pub` modules to make them easier to unpub in the future.
Thoughts on this approach very welcome.

Reply:
What's preventing us from just moving all the code into the solana-client module?

Reply:
If by "all the code" you mean the code in these `temporary_pub` modules, what's preventing us is that the code is being used in the lower-level crates as well.